### DISCUSSION TITLE
Issues for adding header and dll files

### DISCUSSION THREAD

**Discussion by YuanLi-sz** [slmsuite/slmsuite#5] [Q&A] [answered]
Hi,

The current error i am getting when I try to initialize slm is "'slm_funcs' is not defined".
I am using python3.7 and I found in both santec.py and _slm_win.py they are using 'add_dll_directory' and i think python3.7 doesn't support this, so this should be the reason for the error. Does this conclusion seems reasonable to you? And besides switching to python3.8 or newer do you have other suggestions for fixing this issue?

Thank you very much!

**Comment by tpr0p** [comment]
Hi @YuanLi-sz thank you for your note. That conclusion seems reasonable, putting this link here so others can confirm `os.add_dll_directory` was added in py3.8 [0]. The fastest fix would be to upgrade your python version if you are able, or create a new environment with anaconda for using the package [1], else we will look into the proper version < 3.8 way to add dlls and implement it in the package, hopefully in the next day or so.

[0] https://docs.python.org/3/library/os.html#os.add_dll_directory
[1] https://www.anaconda.com/

**Reply by tpr0p** [reply, marked as answer]
see issue #6
